### DISCUSSION TITLE
Using server components on a remote non-nuxt website.

### DISCUSSION THREAD

**Discussion by Youhan** [nuxt/nuxt#23836] [Questions]
I was thinking of building a few components in a nuxt app and then fetching the HTML and styles from another website. Is there a doc or example on that? 

My main issue is how to get the URL for the server component. Currently, it looks like `http://localhost:3000/__nuxt_island/ContentPrivacy_98AD6fIHDv?props=%7B%7D` which looks like a random URL that can be changed after another build?!

@huang-julien Do you think such a use-case is possible at the moment?

**Comment by huang-julien** [comment]
Hey :wave: thank you for pinging me :) . So about non-nuxt remotes server components, there's indeed no documentation atm about how to do it because the API is probably going to change in a close future to ease the usage of non-nuxt remotes islands (and other things).

We basically use `/__nuxt_island/${componentName}_${hashMadeOfPropsAndOtherThings}${query}` on all remotes islands. For now you can parse the request path to know what you need to render.

This means that all `/__nuxt_island/*` routes on your remote server should be reserved for islands.btw the response needs to satisfiy `NuxtIslandResponse` https://github.com/nuxt/nuxt/blob/f4b7406e10223359edd4db58d3d9787c99069f3b/packages/nuxt/src/core/runtime/nitro/renderer.ts#L63

**Reply by Youhan** [reply]
Hi 👋
Thank you for your answer. 

Say, the nuxt app is on `a-nuxt.com` and my other website is using vanilla js is hosted on `vanillajs.com`
- `a-nuxt.com` is a nuxt app that has `/components/PrivacyContent.server.vue`. 
- `vanillajs.com` can do `fetch("a-nuxt.com/__nuxt_island/PrivacyContent")` to get the HTML and styles.

I think if the `PrivacyContent` component is completely static and does not need any props, then fetching `/__nuxt_island/${componentName}` without the hash should be ok?

**Reply by Youhan** [reply]
Basically, I want to build with Nuxt, but use the server component on another app.

**Reply by huang-julien** [reply]
You'll have to pass `https://vanillajs.com` as source to `NuxtIsland` : 
`<NuxtIsland source="https://vanillajs.com" name="PrivacyContent" />`

NuxtIsland will make a call at `/__nuxt_island/PrivacyContent_${hash}`.

So you need to catch all `/__nuxt_island/*` on `vanillajs.com` routes with a single controller
